mlflow/sklearn/__init__.py

```diff
@@ -948,6 +1334,21 @@ def patched_fit(original, self, *args, **kwargs):
                     FLAVOR_NAME, class_def, func_name, patched_fit, manage_run=True,
                 )
 
+        for func_name in ["predict", "transform", "predict_proba"]:
```

Comment:
Do we handle [KMeans.fit_predict()](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.KMeans.html#sklearn.cluster.KMeans.fit_predict)? If not, do you have any ideas about how we can support it? If it's difficult to support and we don't already, can we include documentation to indicate that we don't support it?

Reply:
I think we shouldn't patch `fit_predict`/`fit_transform`
because they will do:
(1) do training model on training dataset
(2) run prediction/transform on training dataset
we can only get training metrics if we patch them.

Reply:
what happens when users take the predictions and feed them to other metrics? We don't necessarily need to handle this now, but it's useful to document that we don't handle it.